pkg/virtual/generic/apiserver/apiserver.go

```diff
@@ -0,0 +1,107 @@
+package apiserver
+
+import (
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apiserver/pkg/registry/rest"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	restclient "k8s.io/client-go/rest"
+)
+
+type RestStorageBuidler func(config CompletedConfig) (rest.Storage, error)
+
+type ExtraConfig struct {
+	KubeAPIServerClientConfig *restclient.Config
+	AdditionalConfig          interface{}
+
+	GroupVersion schema.GroupVersion
+	StorageBuilders map[string] RestStorageBuidler
+
+	// TODO these should all become local eventually
+	Scheme *runtime.Scheme
+	Codecs serializer.CodecFactory
+
+	makeStorage sync.Once
```

Comment:
Generally config holds things that the user passes in, what are these doing in this struct?

Reply:
Removed in https://github.com/davidfestal/kcp/pull/2.